apps/service_broadcast/lib/broadcast/stream/store.ex

```diff
@@ -9,7 +9,20 @@ defmodule Broadcast.Stream.Store do
     Brook.ViewState.merge(@collection, identifier(load), %{"load" => load})
   end
 
-  @spec get!(dataset_id :: String.t(), subset_id :: String.t()) :: Load.Broadcast.t()
+  @spec mark_done(Load.Broadcast.t()) :: :ok
+  def mark_done(%Load.Broadcast{} = load) do
+    Brook.ViewState.merge(@collection, identifier(load), %{"done" => true})
+  end
+
+  @spec done?(Load.Broadcast.t()) :: boolean
+  def done?(%Load.Broadcast{} = load) do
+    case Brook.get!(@instance, @collection, identifier(load)) do
+      nil -> false
+      map -> Map.get(map, "done", false)
+    end
+  end
+
+  @spec get!(dataset_id :: String.t(), subset_id :: String.t()) :: Load.Broadcast.t() | nil
```

Comment:
same question as before? does it make sense to let these keep piling up if they're only there for infrequent references or would we want to create a second collection and migrate completed records to that one?

Reply:
It doesn't feel like the JSR is there to do that right now, but if the team prefers that we could go back and change it everywhere. I think it's a good call out, but I think it could sacrifice simplicity for performance, and I'd like performance to drive that.

Reply:
until we have a paying client with a timeline who's money we're burning through, i feel like the "J" is "the best possible open source project that someone might find valuable". To me, the whole concept of JSR is weighing the pros and cons of an action with respect to a budget and a timeline, neither of which are really applicable here so now's the time to be thinking about architectural choices that might be painful later.

Reply:
if this (or anything) is actually a bad idea, or it's not worth doing within a specific story that's a different matter, but for as long as we're not being paid by a client on a timeline I want to push back on the idea of "action not being worth the effort" especially when it relates to an open source project that, independently of any particular use we may find for it, has no real timeline or budgetary constraints.